pandas/core/algorithms.py

```diff
@@ -2128,3 +2126,26 @@ def sort_mixed(values):
         np.putmask(new_codes, mask, na_sentinel)
 
     return ordered, ensure_platform_int(new_codes)
+
+
+def _sort_mixed(values):
+    """ order ints before strings in 1d arrays, safe in py3 """
+    str_pos = np.array([isinstance(x, str) for x in values], dtype=bool)
+    nums = np.sort(values[~str_pos])
+    strs = np.sort(values[str_pos])
+    return np.concatenate([nums, np.asarray(strs, dtype=object)])
+
+
+def _sort_tuples(values: np.ndarray[tuple]):
```

Comment:
@simonjayhawkins i just noticed this `np.ndarray[tuple]` annotation.  are we ready to start making our ndarray annotations more specific?

Reply:
this is picked up by #36092 `pandas\core\algorithms.py:2217: error: "ndarray" expects no type arguments, but 1 given  [type-arg]`

so will need to be corrected at some point.